### PR TITLE
Expense tracker

### LINKED ISSUE #779: EXPENSE TRACKER USING Tkinter GUI
Aim:
The project would aim to create an app to keep a record of daily expenses and stocks in a proper format so that the user can analyze all the expenditures efficiently ,correctly and manage the funds in a better manner.

Details:

- The app would prompt the user to enter date of the expenditure which will be stored .
-Again the app would prompt the user for a name and other expense details . 
-Once submitted, the user can see the expenses under viewexpense button for each day on the tkinter window.

### PR DESCRIPTION
**Pull request by Amit366**
# Description

Added script for Expense tracker

## Fixes #779 

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

yes

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [ ] My changes generate no new warnings
- [ ] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Add a database to store the data entered by the user

**Comment by Amit366** [comment]
@kaustubhgupta the project is about GUI and not database. So I did only GUI.

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta the project is about GUI and not database. So I did only GUI.

Then how will a user track the expenses? If I enter today's expenses, then how will I compare them in the next week or month? The GUI is just for improving the user interface.

**Comment by Amit366** [comment]
Then should I use Django for database??

**Comment by kaustubhgupta** [comment]
> Then should I use Django for database??

Django for Db? Use SQLite

**Comment by Amit366** [comment]
@kaustubhgupta updated. Since there is a database so I request you to give it a level 3

File: Expense Tracker/script.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+def delete():
+    with db:
+        c = db.cursor()
+        c.execute("DROP table expense")
+        db.commit()
+
+db = sqlite3.connect('expense.db')
+c = db.cursor()
+
+# uncomment this when running the script for the first time or when deleting it
```
Comment (kaustubhgupta) [inline Expense Tracker/script.py:37]: This is not the way of implementing the code "comment-uncomment". Refer to this implementation for databases: https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/899

Also, delete the database uploaded: expense.db

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
> @kaustubhgupta updated. Since there is a database so I request you to give it a level 3

Not possible, we have merged many issues with GUI and db are merged with level2. Some exceptions were there but then their issue was complex. This issue is fairly moderate and level2 is appropriate for it. 

@santushtisharma10 @antrikshmisri What do you think? Should we inc the level?

**Comment by antrikshmisri** [comment]
> > @kaustubhgupta updated. Since there is a database so I request you to give it a level 3
> 
> Not possible, we have merged many issues with GUI and db are merged with level2. Some exceptions were there but then their issue was complex. This issue is fairly moderate and level2 is appropriate for it.
> 
> @santushtisharma10 @antrikshmisri What do you think? Should we inc the level?

I think the level-2 should suffice with the complexity of the project.

File: Expense Tracker/script.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    for row in cursor.fetchall():
+        print(row)
+
+def delete():
+    with db:
+        c = db.cursor()
+        c.execute("DROP table expense")
+        db.commit()
+
+db = sqlite3.connect('expense.db')
```
Comment (kaustubhgupta) [inline Expense Tracker/script.py:34]:
```suggestion
db = sqlite3.connect('./Expense Tracker/expense.db')
```

File: Expense Tracker/script.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        c.execute("INSERT INTO expense(Dates, Items, Expense) VALUES(?,?,?)",(x,y,z))
+       
+
+
+def show():
+    connt=sqlite3.connect('expense.db')
+    cursor=connt.cursor()
+    cursor.execute("SELECT * FROM expense")
+    for row in cursor.fetchall():
+        print(row)
```
Comment (kaustubhgupta) [inline Expense Tracker/script.py:26]: instead of printing, show the results in the table in the GUI

File: Expense Tracker/script.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+def Addexpense():
+    x = Edate.get()
+    y = Title.get()
+    z = Eexpense.get()
+    data = [x,y,z]
+    TVExpense.insert('','end',values=data)
+    with db:
+        c = db.cursor()
+        c.execute("INSERT INTO expense(Dates, Items, Expense) VALUES(?,?,?)",(x,y,z))
```
Comment (kaustubhgupta) [inline Expense Tracker/script.py:17]: handle the case when there is no expense table.

File: Expense Tracker/script.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    connt=sqlite3.connect('expense.db')
+    cursor=connt.cursor()
+    cursor.execute("SELECT * FROM expense")
+    for row in cursor.fetchall():
+        print(row)
+
+def delete():
+    with db:
+        c = db.cursor()
+        c.execute("DROP table expense")
```
Comment (kaustubhgupta) [inline Expense Tracker/script.py:31]: When I click "delete", it should delete that record and not the whole table

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
-

File: Expense Tracker/script.py (hunk 3 above)
Comment (Amit366) [reply Expense Tracker/script.py:26]: In the GUI the items that we add is visible. Then how can I show the entire list??

File: Expense Tracker/script.py (hunk 4 above)
Comment (Amit366) [reply Expense Tracker/script.py:17]: If there is no table then it creates automatically @kaustubhgupta

**Review by kaustubhgupta — APPROVED** [review]
Good job 💯 💯

**Review by antrikshmisri — APPROVED** [review]
LGTM 👍🏻

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]

**Pull request merged**